pkg/config/configobserver/configobserver_test.go

```diff
@@ -100,100 +98,11 @@ func TestChangeSupportConfig(t *testing.T) {
 
 }
 
-// ignore until resolved
-func testChangeObserved(t *testing.T) {
-	setIntervals := map[int]time.Duration{
-		0: time.Duration(10 * time.Minute),
-		1: time.Duration(1 * time.Minute),
-		2: time.Duration(3 * time.Minute),
-		3: time.Duration(4 * time.Minute),
-	}
-
-	klog.SetOutput(utils.NewTestLog(t).Writer())
-
-	ctrl := config.Controller{}
-	kube := kubeClientResponder{}
-	// The initial values set in configobserver.New
-	secs := map[string]*corev1.Secret{
-		pullSecretKey: &corev1.Secret{Data: map[string][]byte{
-			".dockerconfigjson": fakeDockerConfig(),
-		}},
-		supportKey: &corev1.Secret{Data: map[string][]byte{
-			"username":  []byte("someone"),
-			"password":  []byte("secret"),
-			"endpoint":  []byte("http://po.rt"),
-			intervalKey: []byte("10m"),
-		}},
-	}
-
-	provideSecretMock(&kube, secs)
-	// New reads first k8 configuration
-	co := New(ctrl, &kube)
-	// set some initial config because we are tracking changes only
-	co.setConfigLocked(&config.Controller{})
-
-	// observe changes every 50 ms
-	co.checkPeriod = 50 * time.Millisecond
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	// Watch for changes in configurations
-	done := make(chan bool)
-	go co.Start(ctx)
-	changedC, _ := co.ConfigChanged()
-
-	// Sets gather intervals in config to 3 and 4 minutes after 100ms elapses
-	go func() {
-
-		for i := range setIntervals {
-			time.Sleep(100 * time.Millisecond)
-			secs[supportKey].Data[intervalKey] = []byte(setIntervals[i].String())
-		}
-		// Give observer chance to catch the change
-		time.Sleep(50 * time.Millisecond)
-		done <- true
-	}()
-
-	actualIntervals := map[int]time.Duration{}
-	actIntMu := sync.Mutex{}
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				done <- true
-
-			case <-changedC:
-				actInt := co.Config().Interval
-
-				actIntMu.Lock()
-				actIntMu.Unlock()
-				actualIntervals[len(actualIntervals)] = actInt
-			}
-		}
-	}()
-	<-done
-
-	if !reflect.DeepEqual(setIntervals, actualIntervals) {
-		t.Fatalf("the expected intervals didn't match actual intervals. \nExpected %v \nActual %v", setIntervals, actualIntervals)
-	}
-}
-
```

Comment:
Why are you removing all this? I know it's ignored, but I don't see any harm in leaving it there.

Reply:
Unused code should be always removed, we are using version control to handle it to us. So, no need to keep it in the repository, a few enumerated reasons related to code quality:

- "Out of sight" is also "Out of mind";
- Reduce the code base size;
- Version control handles it and keeps it forever;
- Lint will always complain about unused functions, variables, etc;

There are also few articles about it:

https://www.infoq.com/news/2017/02/dead-code/
https://blog.ndepend.com/no-excuse-dead-code/
https://www.oreilly.com/library/view/becoming-a-better/9781491905562/ch04.html

Reply:
OK then it would be good to fix the test or at least create some tracking issue, because "Out of sight" is also "Out of mind" as you said :)

Reply:
I'll create an issue to review this test.